### DISCUSSION TITLE
issue: values on useFieldArray & useWatch are not changed when calling resetField, when there is no register, useController or Controller available

### DISCUSSION THREAD

**Discussion by stijnvanderlaan** [react-hook-form/react-hook-form#8721] [General]
### Version Number

7.33.1

### Codesandbox/Expo snack

https://codesandbox.io/s/usefieldarray-ef1otf?file=/src/App.tsx

### Steps to reproduce

1. Go to 'https://codesandbox.io/s/usefieldarray-ef1otf?file=/src/App.tsx'
2. Click on 'resetField "fields"' or resetField 1
3. The fields should be cleared to 2 items
4. Click on Toggle register
5. Click on 'resetField "fields"' or resetField 1
6. Now everything is updated

### Expected behaviour

I expect the fields to be updated, it's unclear why it's not updated because reset works fine.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
please use UUID or id from the library.

**Reply by stijnvanderlaan** [reply]
Hi [bluebill1049](https://github.com/bluebill1049), 

Thanks for your reply, but as far as know I am using the library id:
```
keyName: "_id"
```

```
key={field._id}
```

Would you care to elaborate on that statement?

**Reply by stijnvanderlaan** [reply]
Hi [bluebill1049](https://github.com/bluebill1049),

Can you please explain? I can use a workaround for `useWatch` (using `useController`), but for `useFieldArray` I need to use `reset`, otherwise the values are not being updated, but I don't want to reset the other values.

**Reply by stijnvanderlaan** [reply]
Oke, I did found a workaround for `useFieldArray` by using adding `useController` for the array.

**Reply by PetterRuud** [reply]
Do you have an example how you used the controller. Did you wrap the whole fields.map()?
I used a hidden field with register for this. Not optimal, but fixed the issue.
